README.md

```diff
@@ -177,44 +177,65 @@ Example of a partial CR:
+
+The decision to make it available as an environment variable is made in the following ascending order of precedence:
 * value of the `bindAs` attribute in the backing service as defined in its [annotations](annotations.md#data-model--building-blocks-for-expressing-binding-information), applying to the binding item referenced by the annotation.
 * value of `ServiceBinding`'s global `bindAs` element, which applies to all binding data.
 * value of the `bindAs` attribute in each of the `dataMappings` elements inside `ServiceBinding`.
 
-#### Injecting data
-
-The key `SERVICE_BINDINGS` acts as a global map of the service bindings and **MUST** always be injected into the environment.  It contains a JSON payload with `bindingKeys` key name containing a list of all available binding information available. Each item of the `bindingKeys` list includes an object containing `name`, `bindAs` and an optional `mountPath` (if it is bound as a volume).
-
-Example:
-
-```json
-SERVICE_BINDINGS = {
-  "bindingKeys": [
-    {
-      "name": "KAFKA_USERNAME",
-      "bindAs": "envVar"
-    },
-    {
-      "name": "KAFKA_PASSWORD",
-      "bindAs": "volume",
-      "mountPath": "/platform/bindings/secret/"
-    }
-  ]
-}
-```
+If a particular binding key is made available as an environment variable an entry **MUST** be made at the following file:
 
-In the example above, the application **MAY** query the environment variable `SERVICE_BINDINGS`, walk its JSON payload and learn that `KAFKA_USERNAME` is available as an environment variable, and that `KAFKA_PASSWORD` is available as a mounted file inside the directory `/platform/bindings/secret/`.
+```
+<mountPathPrefix>/<service-id>/metadata/envVars
+```
 
-#### Mounting data
-Implementations of this specification must bind the following data into the consuming application container:
+Where:
+* `envVars` is a file where each line contains an environment variable name, representing one of the binding properties for this service.  
 
+For example, `platform/bindings/mongodb/metadata/envVars` could have:
 ```
-<mountPathPrefix>/bindings/secret/<persisted_secret>
+MONGODB_HOST
+MONGODB_PORT
 ```
 
-Where:
-* `<mountPathPrefix>` defaults to `platform` if not specified in the `ServiceBinding` CR via the `mountPathPrefix` element.
-* `<persisted_secret>` represents a set of files where the filename is a Secret key and the file contents is the corresponding value of that key.
+#### Guidelines for intermediate binding representation
+
+This specification does not mandate a particular methodology for implementations to process `ServiceBinding` CRs and mount the resulting data as per the requirements set in the previous sections, which allows for different frameworks to have flexibility in exactly how they fulfill the binding request.
+
+However, it is **RECOMMENDED** that implementations expose their intermediate binding representation (i.e. the model containing the binding data prior to mount) in the following way:
```

Comment:
It was in my head, but wasn't written down 😄 

I added a new commit which hopefully clarifies:
```
However, it is RECOMMENDED that implementations expose their intermediate binding representation (i.e. the model containing the binding data prior to mount) in the following way, so that Operator-based deployments or any other processing unit that owns the deployment can take over the responsibility for mounting the data into the container:
```

Reply:
Basically, there are scenarios where an Operator created the deployment, so if another Operator does the Service Binding processing we want a way to just fetch the resulting / processed Secret, and do the mounting via the first Operator., to avoid a reconcile call that will remove the mount.